PWGHF/TableProducer/candidateCreator2Prong.cxx

```diff
@@ -248,17 +272,37 @@ struct HfCandidateCreator2Prong {
     }
   }
 
-  template <bool doPvRefit, typename CandType, typename TTracks>
-  void runCreator2ProngWithKFParticle(aod::Collisions const& collisions,
+  template <bool doPvRefit, int centEstimator, typename Coll, typename CandType, typename TTracks>
+  void runCreator2ProngWithKFParticle(Coll const& collisions,
                                       CandType const& rowsTrackIndexProng2,
                                       TTracks const& tracks,
                                       aod::BCsWithTimestamps const& bcWithTimeStamps)
   {
 
     for (const auto& rowTrackIndexProng2 : rowsTrackIndexProng2) {
+
+      // reject candidates in collisions outside the centrality range
+      auto collision = rowTrackIndexProng2.template collision_as<Coll>();
+      float centrality = -1.;
+      if constexpr (centEstimator != CentralityEstimator::None) {
+        if constexpr (centEstimator == CentralityEstimator::FT0A) {
+          centrality = collision.centFT0A();
+        } else if constexpr (centEstimator == CentralityEstimator::FT0C) {
+          centrality = collision.centFT0C();
+        } else if constexpr (centEstimator == CentralityEstimator::FT0M) {
+          centrality = collision.centFT0M();
+        } else if constexpr (centEstimator == CentralityEstimator::FV0A) {
+          centrality = collision.centFV0A();
+        } else {
+          LOGP(fatal, "Centrality estimator not set!");
+        }
```

Comment:
`FT0A` and `FV0A` can be removed

Reply:
yes, it's done in the last commit

Reply:
uh, I forgot it for the KF particle case, I'll solve it thanks